addons/test_mail/tests/test_mail_thread_internals.py

```diff
@@ -446,14 +444,95 @@ def test_message_get_suggested_recipients(self):
                         'email': self.test_partner.email_normalized,
                         'name': self.test_partner.name,
                         'partner_id': self.test_partner.id,
-                        'reason': 'Customer Email',
                     }
                 )
 
         # do not propose public partners
         ticket_public = self.env['mail.test.ticket.mc'].create({'customer_id': self.user_public.partner_id.id})
         self.assertFalse(ticket_public._message_get_suggested_recipients())
 
+    @users("employee")
+    def test_message_get_suggested_recipients_conversation(self):
+        """ Test suggested recipients in a conversation based on discussion
+        history: email_{cc/to} of previous messages, ... """
+        test_cc_tuples = [
+            ('Test Record Cc', 'test.record.cc@test.example.com'),
+            ('Test Msg Cc', 'test.msg.cc@test.example.com'),
+            ('Test Msg Cc 2', 'test.msg.cc.2@test.example.com'),
+        ]
+        test_to_tuples = [
+            ('Test Msg To', 'test.msg.to@test.example.com'),
+            ('Test Msg To 2', 'test.msg.to.2@test.example.com'),
+        ]
+        test_emails = [x[1] for x in test_cc_tuples] + [x[1] for x in test_to_tuples]
+        self.assertFalse(self.env['res.partner'].search([('email_normalized', 'in', test_emails)]))
+
+        test_record = self.env['mail.test.recipients'].create({
+            'email_cc': tools.mail.formataddr(test_cc_tuples[0]),
+            'name': 'Test Recipients',
+        })
+        messages = self.env['mail.message']
+        for user, post_values in [
+            (self.user_root, {
+                'body': 'First incoming email',
+                'incoming_email_cc': tools.mail.formataddr(test_cc_tuples[1]),
+                'incoming_email_to': tools.mail.formataddr(test_to_tuples[0]),
+                'message_type': 'email',
+            }),
+            (self.user_employee, {
+                'body': 'Salesman reply by email',
+                'incoming_email_cc': tools.mail.formataddr(test_cc_tuples[2]),
+                'incoming_email_to': tools.mail.formataddr(test_to_tuples[1]),
+                'message_type': 'email',
+            }),
+        ]:
+            messages += test_record.with_user(user).message_post(**post_values)
+        self.assertEqual(test_record.message_partner_ids, self.user_employee.partner_id)
+
+        recipients = test_record._message_get_suggested_recipients(reply_message=messages[0], no_create=True)
```

Comment:
if `recipients == []`, the test will be green, can we either check `len(recipients) == 3` or use `strict=True` in zip ?

Reply:
Oh nice, did not know they finally added a strict argument. As it is since 3.10 is it ok to use it :) thanks